presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/JdbcMetadata.java

```diff
@@ -302,6 +304,29 @@ public JdbcTableHandle getTableHandle(ConnectorSession session, SchemaTableName
         return Optional.of(new LimitApplicationResult<>(handle, jdbcClient.isLimitGuaranteed(session)));
     }
 
+    @Override
+    public Optional<TableScanRedirectApplicationResult> applyTableScanRedirect(ConnectorSession session, ConnectorTableHandle table)
+    {
+        JdbcTableHandle tableHandle = (JdbcTableHandle) table;
+
+        List<JdbcColumnHandle> sourceColumns = tableHandle.getColumns()
+                .map(columnSet -> (List<JdbcColumnHandle>) columnSet.stream().map(JdbcColumnHandle.class::cast).collect(toImmutableList()))
+                .orElseGet(() -> jdbcClient.getColumns(session, tableHandle));
+
+        Optional<JdbcTableScanRedirectionResult> result = jdbcClient.getRedirectedTableScan(session, tableHandle, sourceColumns);
```

Comment:
passing of `sourceColumns` is redundant. `getRedirectedTableScan` can extract these itself.

Reply:
True, though doing it here avoids repeating that code in jdbc connectors.